web/router.go

```diff
@@ -104,6 +83,95 @@ func v2Routes(app *services.ChainlinkApplication, engine *gin.Engine) {
 	v2.GET("/config", cc.Show)
 }
 
+func guiAssetRoutes(engine *gin.Engine) {
+	box := NewBox()
+	boxList := box.List()
+
+	engine.NoRoute(func(c *gin.Context) {
+		path := c.Request.URL.Path
+		matchedBoxPath := matchPath(boxList, path)
+
+		if matchedBoxPath == "" {
+			if filepath.Ext(path) == "" {
+				matchedBoxPath = wildcardMatchPath(
+					boxList,
+					path,
+					"index.html",
+				)
+			} else if filepath.Ext(path) == ".json" {
+				matchedBoxPath = wildcardMatchPath(
+					boxList,
+					filepath.Dir(path),
+					filepath.Base(path),
+				)
+			}
+		}
+
+		if matchedBoxPath != "" {
+			file, err := box.Open(matchedBoxPath)
+			if err != nil {
+				if err == os.ErrNotExist {
+					c.AbortWithStatus(http.StatusNotFound)
+				} else {
+					err := fmt.Errorf("failed to open static file '%s': %+v", path, err)
+					logger.Error(err.Error())
+					c.AbortWithError(http.StatusInternalServerError, err)
+				}
+				return
+			}
+
+			http.ServeContent(c.Writer, c.Request, path, time.Time{}, file)
+		}
+	})
+}
+
+func wildcardMatchPath(boxList []string, path string, file string) (matchedPath string) {
```

Comment:
This is a big function with no side effects and no tests, I definitely wanna see some tests here.

Reply:
Tests are here https://github.com/smartcontractkit/chainlink/pull/384/files#diff-167bd5a8e552f506ed6d917eef0c9ec0R1.

`TestGuiAssets_WildcardIndexHtml` & `TestGuiAssets_WildcardRouteInfo`

Reply:
No unit tests?